projects/packages/publicize/src/class-keyring-service-helper.php

```diff
@@ -0,0 +1,341 @@
+	 *
+	 * @var bool
+	 */
+	private static $is_sharing_page_registered = false;
+
+	/**
+	 * Initialize instance.
+	 *
+	 * @since $$next-version$$
+	 */
+	public static function init() {
+		if ( self::$instance === null ) {
+			self::$instance = new Jetpack_Keyring_Service_Helper();
+		}
+
+		return self::$instance;
+	}
+
+	const SERVICES = array(
+		'facebook'                 => array(
+			'for' => 'publicize',
+		),
+		'twitter'                  => array(
+			'for' => 'publicize',
+		),
+		'linkedin'                 => array(
+			'for' => 'publicize',
+		),
+		'tumblr'                   => array(
+			'for' => 'publicize',
+		),
+		'path'                     => array(
+			'for' => 'publicize',
+		),
+		'google_plus'              => array(
+			'for' => 'publicize',
+		),
+		'google_site_verification' => array(
+			'for' => 'other',
+		),
+	);
+
+	/**
+	 * Constructor
+	 *
+	 * @since $$next-version$$
+	 */
+	private function __construct() {
+		add_action( 'admin_menu', array( __CLASS__, 'register_sharing_page' ) );
+
+		add_action( 'load-settings_page_sharing', array( __CLASS__, 'admin_page_load' ), 9 );
+	}
+
+	/**
+	 * We need a `sharing` page to be able to connect and disconnect services.
+	 *
+	 * @since $$next-version$$
+	 */
+	public static function register_sharing_page() {
+		if ( self::$is_sharing_page_registered ) {
+			return;
+		}
+
+		self::$is_sharing_page_registered = true;
+
+		if ( ! current_user_can( 'manage_options' ) ) {
+			return;
+		}
+
+		global $_registered_pages;
+
+		require_once ABSPATH . 'wp-admin/includes/plugin.php';
+
+		$hookname = get_plugin_page_hookname( 'sharing', 'options-general.php' );
+		add_action( $hookname, array( __CLASS__, 'admin_page_load' ) );
+		$_registered_pages[ $hookname ] = true; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+	}
+
+	/**
+	 * Return a list of services.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $filter Choose 'all' to get all connected services, vs. just the connected ones.
+	 */
+	public function get_services( $filter = 'all' ) {
+		$services = array();
+
+		if ( 'all' === $filter ) {
+			return $services;
+		} else {
+			$connected_services = array();
+			foreach ( $services as $service => $empty ) {
+				$connections = $this->get_connections( $service );
+				if ( $connections ) {
+					$connected_services[ $service ] = $connections;
+				}
+			}
+			return $connected_services;
+		}
+	}
+
+	/**
+	 * Gets a URL to the public-api actions. Works like WP's admin_url.
+	 * On WordPress.com this is/calls Keyring::admin_url.
+	 *
+	 * @since $$next-version$$
+	 *
+	 * @param string $service Shortname of a specific service.
+	 * @param array  $params  Parameters to append to an API connection URL.
+	 *
+	 * @return URL to specific public-api process
+	 */
+	private static function api_url( $service = false, $params = array() ) {
+		/**
+		 * Filters the API URL used to interact with WordPress.com.
+		 *
+		 * @since 2.0.0
```

Comment:
Since that comes from Jetpack, we'll want to update those references to something like `@since-jetpack 2.0.0`.